src/OpenLoco/src/GameCommands/Track/RemoveTrack.cpp

```diff
@@ -0,0 +1,295 @@
+#include "RemoveTrack.h"
+#include "Economy/Economy.h"
+#include "Map/RoadElement.h"
+#include "Map/SignalElement.h"
+#include "Map/TileManager.h"
+#include "Map/Track/TrackData.h"
+#include "Map/TrackElement.h"
+#include "Objects/BridgeObject.h"
+#include "Objects/TrackExtraObject.h"
+#include "Objects/TrackObject.h"
+#include "Random.h"
+#include "RemoveSignal.h"
+#include "RemoveTrainStation.h"
+#include "Vehicles/Vehicle.h"
+#include "ViewportManager.h"
+
+namespace OpenLoco::GameCommands
+{
+    static World::TrackElement* getElTrackAt(const TrackRemovalArgs& args, const uint8_t flags, const World::Pos3 pos, const uint8_t sequenceIndex)
+    {
+        auto tile = World::TileManager::get(pos);
+        for (auto& el : tile)
+        {
+            auto* elTrack = el.as<World::TrackElement>();
+            if (elTrack == nullptr)
+            {
+                continue;
+            }
+            if (elTrack->rotation() != args.rotation)
```

Comment:
there is no check for height in this getElTrackAt. I'm not sure if that is correct.